packages/java-packages/snippet-replacer-maven-plugin/src/main/java/azuresdk/plugin/SnippetReplacer.java

```diff
@@ -0,0 +1,408 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.*;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.*;
+import java.util.regex.*;
+
+import com.google.common.base.Verify;
+import org.apache.maven.plugin.Mojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.logging.Log;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.stream.Collectors;
+
+public class SnippetReplacer {
+    private final static Pattern SNIPPET_DEF_BEGIN = Pattern.compile("\\s*\\/\\/\\s*BEGIN\\:\\s+([a-zA-Z0-9\\.\\#\\-\\_]*)\\s*");
+    private final static Pattern SNIPPET_DEF_END = Pattern.compile("\\s*\\/\\/\\s*END\\:\\s+([a-zA-Z0-9\\.\\#\\-\\_]*)\\s*");
+    private final static Pattern SNIPPET_CALL_BEGIN = Pattern.compile("(\\s*)\\*?\\s*<!--\\s+src_embed\\s+([a-zA-Z0-9\\.\\#\\-\\_]*)\\s*-->");
+    private final static Pattern SNIPPET_CALL_END = Pattern.compile("(\\s*)\\*?\\s*<!--\\s+end\\s+([a-zA-Z0-9\\.\\#\\-\\_]*)\\s*-->");
+    private final static Pattern WHITESPACE_EXTRACTION = Pattern.compile("(\\s*)(.*)");
+
+    private static PathMatcher SAMPLE_PATH_GLOB = FileSystems.getDefault().getPathMatcher("glob:**/src/samples/java/**/*.java");
+    private static PathMatcher JAVA_GLOB = FileSystems.getDefault().getPathMatcher("glob:**/*.java");
+
+    private static HashMap<String, String> REPLACEMENT_SET = new HashMap<String, String>(){{
+        put("\"", "&quot;");
+        put(">", "&#62;");
+        put("<", "&#60;");
+        put("@", "{@literal @}");
+        put("{", "&#123;");
+        put("}", "&#125;");
+        put("(", "&#40;");
+        put(")", "&#41;");
+        put("/", "&#47;");
+        put("\\", "&#92;");
+    }};
+
+    public SnippetReplacer(){}
+
+    public SnippetReplacer(String mode, File folderToVerify, Log logger) throws MojoExecutionException, IOException {
+        switch (mode) {
+            case "update":
+                this.runUpdate(folderToVerify);
+                break;
+            case "verify":
+                this.runVerification(folderToVerify, logger);
+                break;
+            default:
+                throw new MojoExecutionException(String.format("Unrecognized snippet-replacer mode: %s.", mode));
+        }
+    }
+
+    public void runVerification(File folderToVerify, Log logger) throws IOException, MojoExecutionException{
+        List<Path> allLocatedJavaFiles = glob(folderToVerify.toPath(), JAVA_GLOB);
+        List<Path> snippetSources = globFiles(allLocatedJavaFiles, SAMPLE_PATH_GLOB);
+        List<VerifyResult> discoveredIssues = new ArrayList<VerifyResult>();
+        HashMap<String, List<String>> foundSnippets = new HashMap<String, List<String>>();
+
+        // scan the sample files for all the snippet files
+        foundSnippets = this.getAllSnippets(snippetSources);
+
+        // walk across all the java files, run UpdateSrcSnippets
+        for(Path sourcePath: allLocatedJavaFiles){
```

Comment:
minor nits: 
```suggestion
        for (Path sourcePath : allLocatedJavaFiles) {
```

Reply:
Will make a pass for all the foreach loops on this too.